src/algebra/gcd_domain.lean

```diff
@@ -27,41 +27,108 @@ attribute [simp] norm_unit_coe_units norm_unit_zero norm_unit_mul
 section normalization_domain
 variable [normalization_domain α]
 
+def normalize (x : α) : α :=
+x * norm_unit x
+
+theorem associated_normalize {x : α} : associated x (normalize x) :=
+⟨_, rfl⟩
+
+theorem normalize_associated {x : α} : associated (normalize x) x :=
+associated_normalize.symm
+
 @[simp] theorem norm_unit_one : norm_unit (1:α) = 1 :=
 norm_unit_coe_units 1
 
+variables (α)
```

Comment:
This should be implicit

Reply:
But it creates metavariables.

Reply:
I agree with chris. Use `rw` if `\t` gives you trouble

Reply:
done